diff -ur -N a/src/etc/rc.initial b/src/etc/rc.initial
--- a/src/etc/rc.initial	2025-12-23 18:08:15.848759000 +1100
+++ b/src/etc/rc.initial	2025-12-26 14:05:37.977282000 +1100
@@ -63,6 +63,7 @@
 	sshd_option='Enable'
 fi
 
+
 # display console menu
 echo ""
 echo " 0) Logout (SSH only)                  9) pfTop"
@@ -82,6 +83,7 @@
 # see what the user has chosen
 case ${opmode} in
 0)
+	[ -e /root/.bash_logout ] && . /root/.bash_logout
 	exit && exit && logout
 	;;
 1)
@@ -106,7 +108,15 @@
 	/etc/rc.initial.ping
 	;;
 8)
-	/bin/tcsh
+	if [ -x /usr/local/bin/bash ]
+	then
+		/usr/local/bin/bash
+	elif [ -x /bin/tcsh ]
+	then
+		/bin/tcsh
+	else
+		/bin/sh
+	fi
 	;;
 9)
 	/usr/local/sbin/pftop
